### ISSUE TITLE
Paying your own invoice crashes clightning

### ISSUE THREAD

**Issue by Kukks** [ElementsProject/lightning#4688]
### Issue and Steps to Reproduce
* Create a BOLT 11 invoice with lightning
* attempt to pay it with the same node.
* lightning container crashes

```
 ./bitcoin-lightning-cli.sh pay BOLT11
lightning-cli: reading response: socket closed
```

Logs:
```
2021-07-28T09:21:32.411632772Z 2021-07-28T09:21:32.411Z INFO    plugin-pay: cmd 68 partid 0: Payment fee constraint 5msat is below exemption threshold, allowing a maximum fee of 5000msat
2021-07-28T09:21:32.414766444Z 2021-07-28T09:21:32.414Z INFO    gossipd: find_route: this is 035736692722a47389d6be965d73a50b1ba63233a1b975aa55db7f90446f5708cb, refusing to create empty route
2021-07-28T09:21:32.601199032Z pay: channel_update 652387x929x1 ignored: fee 5000/5000000 cltv 1000 too large
2021-07-28T09:21:32.601262141Z pay: channel_update 653015x491x0 ignored: fee 5000/5000000 cltv 1000 too large
2021-07-28T09:21:32.604757872Z pay: channel_update 681430x896x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.607569951Z pay: channel_update 682141x2216x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.609510851Z pay: channel_update 670476x1038x1 ignored: fee 5000/5000000 cltv 1000 too large
2021-07-28T09:21:32.609618710Z pay: channel_update 685413x687x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.609630732Z pay: channel_update 691023x1510x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.610343733Z pay: channel_update 679729x939x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.610915616Z pay: channel_update 681034x1664x0 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.611056723Z pay: channel_update 681036x1058x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.611084695Z pay: channel_update 681034x1665x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.611381995Z pay: channel_update 681036x1059x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.614510941Z pay: channel_update 682346x179x4 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.614714618Z pay: channel_update 681612x1659x0 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.615021202Z pay: channel_update 672382x611x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.615165923Z pay: channel_update 673353x201x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.615837451Z pay: channel_update 689118x2618x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.623969686Z pay: channel_update 667866x1720x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624066698Z pay: channel_update 667348x1364x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624079946Z pay: channel_update 667525x624x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624086693Z pay: channel_update 667525x1329x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624092935Z pay: channel_update 667679x860x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624113354Z pay: channel_update 675351x1673x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624122224Z pay: channel_update 684896x1587x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624129406Z pay: channel_update 686408x2163x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624136314Z pay: channel_update 687409x1740x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624143282Z pay: channel_update 687454x1224x7 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.624568046Z pay: channel_update 667849x839x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.628224463Z pay: channel_update 674862x713x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.629278829Z pay: channel_update 682145x1036x2 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.635562500Z pay: channel_update 659219x534x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.635740844Z pay: channel_update 685135x362x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.636580006Z pay: channel_update 678483x1403x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.637219383Z pay: channel_update 651160x1075x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.638185915Z pay: channel_update 649696x1518x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.638407064Z pay: channel_update 649696x1522x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.639741973Z pay: channel_update 682629x761x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.642539041Z pay: channel_update 681574x1661x13 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.648884488Z pay: channel_update 691539x1985x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.650934011Z pay: channel_update 691261x1244x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.660323135Z pay: channel_update 679005x1911x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.663802832Z pay: channel_update 678099x1104x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.663854987Z pay: channel_update 678273x1058x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.664929719Z pay: channel_update 577477x1902x1 ignored: fee 7/6000000 cltv 4 too large
2021-07-28T09:21:32.666109908Z pay: channel_update 681486x1339x4 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.666459812Z pay: channel_update 682103x1859x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.669762225Z pay: channel_update 685553x489x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.670754167Z pay: channel_update 662204x1682x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.671265863Z pay: channel_update 688696x1713x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.671533263Z pay: channel_update 686232x1239x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.672041792Z pay: channel_update 690313x1392x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.672068948Z pay: channel_update 683850x852x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.677065655Z pay: channel_update 681220x1639x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.681565683Z pay: channel_update 679110x1888x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.681985907Z pay: channel_update 667499x276x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.684715915Z pay: channel_update 687434x1755x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.693507723Z pay: channel_update 689009x899x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.693571218Z pay: channel_update 577480x1297x0 ignored: fee 7/6000000 cltv 4 too large
2021-07-28T09:21:32.698729221Z pay: channel_update 689024x552x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.699663835Z pay: channel_update 682173x1367x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.711236614Z pay: channel_update 689022x2103x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.714013481Z pay: channel_update 662882x675x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.716219704Z pay: channel_update 660091x1985x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.716244841Z pay: channel_update 659972x1779x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.716295271Z pay: channel_update 647517x1724x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.716815441Z pay: channel_update 681586x1392x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.722328055Z pay: channel_update 576683x1919x0 ignored: fee 7/6000000 cltv 4 too large
2021-07-28T09:21:32.723600786Z pay: channel_update 577590x1455x0 ignored: fee 7/6000000 cltv 4 too large
2021-07-28T09:21:32.740372343Z pay: channel_update 687426x966x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.740559545Z pay: channel_update 690055x200x1 ignored: fee 99999000/99999 cltv 40 too large
2021-07-28T09:21:32.743259755Z pay: channel_update 683537x141x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.743293212Z pay: channel_update 686434x1266x3 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.743302022Z pay: channel_update 687085x37x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.743677872Z pay: channel_update 661274x1390x3 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.744916182Z pay: channel_update 687454x1706x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.748451383Z pay: channel_update 686735x1229x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.754328457Z pay: channel_update 667348x1365x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.754527901Z pay: channel_update 678017x1242x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.754557900Z pay: channel_update 680828x2106x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.754567718Z pay: channel_update 681440x1099x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.754957910Z pay: channel_update 667501x1651x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.756045760Z pay: channel_update 661166x1802x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.756408393Z pay: channel_update 686735x1261x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.756516548Z pay: channel_update 673831x2160x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.756879872Z pay: channel_update 679041x2355x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.809919559Z pay: channel_update 644616x1401x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.809975225Z pay: channel_update 657322x2367x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.809983905Z pay: channel_update 657764x948x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.809990270Z pay: channel_update 667509x267x1 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.809996415Z pay: channel_update 668001x458x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.811996401Z pay: channel_update 677030x1108x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.823749502Z pay: channel_update 667330x1462x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.823943460Z pay: channel_update 667525x1325x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.825646847Z pay: channel_update 691772x1452x1 ignored: fee 1000000000/1 cltv 40 too large
2021-07-28T09:21:32.825804388Z pay: channel_update 681537x1225x1 ignored: fee 1000/2000000 cltv 40 too large
2021-07-28T09:21:32.827082311Z pay: channel_update 692922x826x0 ignored: fee 0/567171199 cltv 34 too large
2021-07-28T09:21:32.827451375Z pay: channel_update 682096x2219x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.828328234Z pay: channel_update 684826x1079x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.830461368Z pay: channel_update 679707x1813x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.830492178Z pay: channel_update 672677x288x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.830504298Z pay: channel_update 672683x1486x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.830610587Z pay: channel_update 670894x2665x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.834452212Z pay: channel_update 670893x1406x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.834489429Z pay: channel_update 670893x1397x1 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.835363991Z pay: channel_update 684686x1304x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.836544178Z pay: channel_update 680876x1886x1 ignored: fee 3000/800000000 cltv 200 too large
2021-07-28T09:21:32.836612584Z pay: channel_update 687408x2359x0 ignored: fee 0/500000000 cltv 100 too large
2021-07-28T09:21:32.839155343Z pay: channel_update 683675x808x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.839478649Z pay: channel_update 690740x1360x1 ignored: fee 99999000/99999 cltv 40 too large
2021-07-28T09:21:32.839909093Z pay: channel_update 683500x409x1 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.840708617Z pay: channel_update 680255x2058x1 ignored: fee 1000/1055130 cltv 40 too large
2021-07-28T09:21:32.841555137Z pay: channel_update 691924x749x0 ignored: fee 1410065408/30000000 cltv 40 too large
2021-07-28T09:21:32.842284528Z pay: channel_update 635504x1333x1 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.842321055Z pay: channel_update 670647x699x1 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.842329784Z pay: channel_update 683772x2463x1 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.842336538Z pay: channel_update 616326x739x0 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.842343091Z pay: channel_update 592927x1329x0 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.842641227Z pay: channel_update 663096x1104x0 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.842681515Z pay: channel_update 689720x834x9 ignored: fee 141754369/332 cltv 40 too large
2021-07-28T09:21:32.843973297Z pay: channel_update 660112x466x0 ignored: fee 1000/1000000000 cltv 37 too large
2021-07-28T09:21:32.849233561Z pay: channel_update 577555x1983x0 ignored: fee 7/6000000 cltv 4 too large
2021-07-28T09:21:32.849648052Z pay: channel_update 683120x1897x0 ignored: fee 1000000000/10000 cltv 40 too large
2021-07-28T09:21:32.980105132Z 2021-07-28T09:21:32.979Z INFO    plugin-pay: cmd 68 partid 0: Initial limit on max HTLCs: 255, Destination 035736692722a47389d6be965d73a50b1ba63233a1b975aa55db7f90446f5708cb has 17 channels, assuming 15 HTLCs per channel
2021-07-28T09:21:34.069471019Z 2021-07-28T09:21:34.069Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002a0c2955cadf7b83e2bc10a83d35569a6d1b8860885613 0 exited with status 1
2021-07-28T09:21:34.408270925Z 2021-07-28T09:21:34.408Z UNUSUAL plugin-bcli: bitcoin-cli -datadir=/etc/bitcoin -rpcconnect=bitcoind getblock 0000000000000000002a0c2955cadf7b83e2bc10a83d35569a6d1b8860885613 0 exited with status 1
2021-07-28T09:21:35.057190219Z 2021-07-28T09:21:35.056Z INFO    03bc9337c7a28bb784d67742ebedd30a93bacdf7e4ca16436ef3798000242b2251-chan#723: htlc 237 failed from 0th node with code 0x1007 (WIRE_TEMPORARY_CHANNEL_FAILURE)
2021-07-28T09:21:35.061175823Z 2021-07-28T09:21:35.060Z INFO    plugin-pay: cmd 68 partid 1: failed: WIRE_TEMPORARY_CHANNEL_FAILURE (reply from remote)
2021-07-28T09:21:35.186703013Z pay: FATAL SIGNAL 6 (version basedon-v0.9.3-1)
2021-07-28T09:21:35.237458150Z 0x56426a2d017a send_backtrace
2021-07-28T09:21:35.237513630Z  common/daemon.c:38
2021-07-28T09:21:35.237523882Z 0x56426a2d0204 crashdump
2021-07-28T09:21:35.237531201Z  common/daemon.c:51
2021-07-28T09:21:35.237537241Z 0x7f6a7d26e72f ???
2021-07-28T09:21:35.237543374Z  ???:0
2021-07-28T09:21:35.238577606Z 0x7f6a7d0b37bb ???
2021-07-28T09:21:35.238603892Z  ???:0
2021-07-28T09:21:35.238611658Z 0x7f6a7d09e534 ???
2021-07-28T09:21:35.238618221Z  ???:0
2021-07-28T09:21:35.241712363Z 0x56426a2c137e payment_route_fee
2021-07-28T09:21:35.241745114Z  plugins/libplugin-pay.c:428
2021-07-28T09:21:35.241755628Z 0x56426a2c2dc2 payment_getroute
2021-07-28T09:21:35.241839667Z  plugins/libplugin-pay.c:809
2021-07-28T09:21:35.241851753Z 0x56426a2c310f payment_continue
2021-07-28T09:21:35.241858876Z  plugins/libplugin-pay.c:1903
2021-07-28T09:21:35.241864371Z 0x56426a2c58a2 adaptive_splitter_cb
2021-07-28T09:21:35.241869099Z  plugins/libplugin-pay.c:3486
2021-07-28T09:21:35.241873736Z 0x56426a2c30e1 payment_continue
2021-07-28T09:21:35.241878221Z  plugins/libplugin-pay.c:1895
2021-07-28T09:21:35.241883052Z 0x56426a2c4b8c retry_step_cb
2021-07-28T09:21:35.241888298Z  plugins/libplugin-pay.c:2045
2021-07-28T09:21:35.241894448Z 0x56426a2c30e1 payment_continue
2021-07-28T09:21:35.241900346Z  plugins/libplugin-pay.c:1895
2021-07-28T09:21:35.241906728Z 0x56426a2c3da5 waitblockheight_cb
2021-07-28T09:21:35.241912696Z  plugins/libplugin-pay.c:3054
2021-07-28T09:21:35.241918841Z 0x56426a2c30e1 payment_continue
2021-07-28T09:21:35.241924896Z  plugins/libplugin-pay.c:1895
2021-07-28T09:21:35.241930791Z 0x56426a2c471e presplit_cb
2021-07-28T09:21:35.241936788Z  plugins/libplugin-pay.c:3223
2021-07-28T09:21:35.241942617Z 0x56426a2c30e1 payment_continue
2021-07-28T09:21:35.241948470Z  plugins/libplugin-pay.c:1895
2021-07-28T09:21:35.241954438Z 0x56426a2c3ded payee_incoming_limit_step_cb
2021-07-28T09:21:35.241960476Z  plugins/libplugin-pay.c:3581
2021-07-28T09:21:35.241966761Z 0x56426a2c30e1 payment_continue
2021-07-28T09:21:35.241972937Z  plugins/libplugin-pay.c:1895
2021-07-28T09:21:35.241979366Z 0x56426a2c3863 routehint_step_cb
2021-07-28T09:21:35.241985725Z  plugins/libplugin-pay.c:2560
2021-07-28T09:21:35.241992067Z 0x56426a2c30e1 payment_continue
2021-07-28T09:21:35.241998100Z  plugins/libplugin-pay.c:1895
2021-07-28T09:21:35.242004435Z 0x56426a2c53ff shadow_route_listchannels
2021-07-28T09:21:35.242010577Z  plugins/libplugin-pay.c:2858
2021-07-28T09:21:35.243034792Z 0x56426a2be1cb handle_rpc_reply
2021-07-28T09:21:35.243060637Z  plugins/libplugin.c:564
2021-07-28T09:21:35.243069778Z 0x56426a2be2bd rpc_read_response_one
2021-07-28T09:21:35.243076433Z  plugins/libplugin.c:711
2021-07-28T09:21:35.243082863Z 0x56426a2be373 rpc_conn_read_response
2021-07-28T09:21:35.243089514Z  plugins/libplugin.c:731
2021-07-28T09:21:35.243289905Z 0x56426a2e10b3 next_plan
2021-07-28T09:21:35.243314814Z  ccan/ccan/io/io.c:59
2021-07-28T09:21:35.243324508Z 0x56426a2e1586 do_plan
2021-07-28T09:21:35.243331142Z  ccan/ccan/io/io.c:407
2021-07-28T09:21:35.243337758Z 0x56426a2e15b8 io_ready
2021-07-28T09:21:35.243410791Z  ccan/ccan/io/io.c:417
2021-07-28T09:21:35.243740634Z 0x56426a2e2e70 io_loop
2021-07-28T09:21:35.243767994Z  ccan/ccan/io/poll.c:445
2021-07-28T09:21:35.243776630Z 0x56426a2bee9c plugin_main
2021-07-28T09:21:35.243782839Z  plugins/libplugin.c:1414
2021-07-28T09:21:35.246302166Z 0x56426a2bba16 main
2021-07-28T09:21:35.246327658Z  plugins/pay.c:2175
2021-07-28T09:21:35.246335024Z 0x7f6a7d0a009a ???
2021-07-28T09:21:35.246340715Z  ???:0
2021-07-28T09:21:35.246346444Z 0x56426a2b6df9 ???
2021-07-28T09:21:35.246351888Z  ???:0
2021-07-28T09:21:35.246357412Z 0xffffffffffffffff ???
2021-07-28T09:21:35.246362954Z  ???:0
2021-07-28T09:21:35.246368555Z pay: FATAL SIGNAL 11 (version basedon-v0.9.3-1)
2021-07-28T09:21:35.246376885Z 0x56426a2d017a send_backtrace
2021-07-28T09:21:35.246382632Z  common/daemon.c:38
2021-07-28T09:21:35.246388382Z 0x56426a2d0204 crashdump
2021-07-28T09:21:35.246394932Z  common/daemon.c:51
2021-07-28T09:21:35.246400631Z 0x7f6a7d26e72f ???
2021-07-28T09:21:35.246406236Z  ???:0
2021-07-28T09:21:35.246411533Z 0x0 ???
2021-07-28T09:21:35.246417227Z  ???:0
2021-07-28T09:21:35.315696550Z 2021-07-28T09:21:35.315Z INFO    plugin-pay: Killing plugin: Plugin exited before completing handshake.
2021-07-28T09:21:35.315745408Z 2021-07-28T09:21:35.315Z **BROKEN** plugin-pay: Plugin marked as important, shutting down lightningd!
2021-07-28T09:21:35.348676398Z 2021-07-28T09:21:35.348Z INFO    plugin-autoclean: Killing plugin: Plugin exited before completing handshake.
2021-07-28T09:21:35.349555348Z 2021-07-28T09:21:35.349Z INFO    plugin-bcli: Killing plugin: Plugin exited before completing handshake.
2021-07-28T09:21:35.350633258Z 2021-07-28T09:21:35.350Z INFO    plugin-keysend: Killing plugin: Plugin exited before completing handshake.
2021-07-28T09:21:35.351668306Z 2021-07-28T09:21:35.351Z INFO    plugin-txprepare: Killing plugin: Plugin exited before completing handshake.
2021-07-28T09:21:35.356484716Z 2021-07-28T09:21:35.356Z INFO    plugin-spenderp: Killing plugin: Plugin exited before completing handshake.
```

**Comment by vincenzopalazzo** [comment]
@Kukks What is your version of c-lightning? the last version on the master branch?

**Comment by Kukks** [comment]
> @Kukks What is your version of c-lightning? the last version on the master branch?

I believe it is 0.9.3

**Comment by vincenzopalazzo** [comment]
it looks like to be a recursive payment, and @cdecker fixed a couple of bugs in the pay plugin i this new version that will be out soon. It is this also the case?

Maybe upgrade to the new version will fix your problem.

Let wait for the Christian's feedback :)

**Comment by rustyrussell** [comment]
Yep, this has been fixed (actually was fixed in 0.10.0):

Changelog:

 - pay: Payments with an empty route (self-payment) are now aborted. (#4379)
